rest-server/src/templates/yarnContainerScript.mustache

```diff
@@ -69,6 +69,90 @@ docker login --username ${cred[1]} --password ${cred[2]} ${cred[0]} \
   || { echo 'Authorized failed' ; exit 1; }
 {{/ jobData.authFile }}
 
+
+# Get container host list from framework launcher webservice
+function get_host_list()
+{
+  while true; do
+    taskrole_statuses=$(curl -X GET -H "Accept: application/json" {{{ statusUri }}} | jq -r ".aggregatedFrameworkStatus.aggregatedTaskRoleStatuses")
+    jq -re ".[].taskStatuses.taskStatusArray[].containerIp") <<< $(cat $taskrole_statuses) > /dev/null
+    if [ $? -ne 0 ]; then
+      break
+    fi
+    sleep $[ ( $RANDOM % 10 )  + 10 ]s
```

Comment:
>10 [](start = 33, length = 2)

Sleep around 300s

Reply:
We can't let user to wait 5,10,15mins to start a container. But we can do exponential backoff. 5, 10, 20,40, 80 seconds. how about this?

---
In reply to: [184282323](https://github.com/Microsoft/pai/pull/467#discussion_r184282323) [](ancestors = 184282323)

Reply:
5min can be ignored if compared with DL jobs whole running time. Before Launcher support replica to store these status on HDFS, I suggest 5min

---
In reply to: [184283036](https://github.com/Microsoft/pai/pull/467#discussion_r184283036) [](ancestors = 184283036,184282323)

Reply:
And the web server status update interval is 1min now, so wait less than 1min is wasting.

---
In reply to: [184363281](https://github.com/Microsoft/pai/pull/467#discussion_r184363281) [](ancestors = 184363281,184283036,184282323)

Reply:
how about we set it to 60s initially, and 120s the second time, and up to 5mins. Not all jobs are deep learning training jobs. and not all deep learning jobs last a long time.  By the way, we might think of how to speed up the status update. (like deploy a different zk server)

---
In reply to: [184363757](https://github.com/Microsoft/pai/pull/467#discussion_r184363757) [](ancestors = 184363757,184363281,184283036,184282323)

Reply:
ok

---
In reply to: [184366762](https://github.com/Microsoft/pai/pull/467#discussion_r184366762) [](ancestors = 184366762,184363757,184363281,184283036,184282323)